### PR TITLE
Feature/notificiation api

### PR DESCRIPTION
**Pull request by John1s**
I added an API module to get notifications and notification count.
I used a notitification count call so that the UI doesn't need to pull all notifications just to count them.
I changed the UI so that notifications are data driven and added a onclick to dismiss notifications.

Enjoy.

### CODE REVIEW

File: api/Notification.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
+'use strict';
+const {ApiBase} = require('./ApiBase');
+
+
+class NotificationApi extends ApiBase {
```
Comment (kwierchris) [inline api/Notification.js]: I'd like to have this class name and the requires as "Notifications".  Just do it the same way as it is with "Users".

File: api/index.js [hunk 2]
```diff
@@ -1,5 +1,6 @@
 import Auth from './Auth';
 import Users from './Users';
+import NotificationApi from './Notification';
```
Comment (kwierchris) [inline api/index.js]: "Notifications" please.

File: components/MainMenu/MainMenu.react.js [hunk 3]
```diff
@@ -5,6 +5,7 @@ import { ResetTimer } from '../../system';
 import { Menu, MenuIcon, MenuDropdownGrid, Notifications, Notification, MenuAccount, MenuSelect } from '@opuscapita/react-navigation';
 import translations from './i18n';
 import navItems from './data/navItems';
+import {NotificationApi} from '../../api';
```
Comment (kwierchris) [inline components/MainMenu/MainMenu.react.js]: "Notifications" please.

File: components/MainMenu/MainMenu.react.js [hunk 4]
```diff
@@ -44,8 +45,23 @@ class MainMenu extends ConditionalRenderComponent
 
         this.logoImage = 'data:image/svg+xml,' + encodeURIComponent(require('!!raw-loader!./img/oc-logo-white.svg'));
         this.searchTimer = new ResetTimer();
+
+        this.notificationApi = new NotificationApi();
+        this.loadNotifications();
+    }
+
+
+    loadNotifications(){
```
Comment (kwierchris) [inline components/MainMenu/MainMenu.react.js]: Please add some spaces between commas and operators as it is done in the other methods. I really don't like this kind of dense programming and it doesn't make it more readable.

File: components/MainMenu/MainMenu.react.js (hunk 3 above)
Comment (John1s) [reply components/MainMenu/MainMenu.react.js]: You already have Notifications from the react-navigation module which is why I needed a name that could distinguish between the UI and API components.

File: components/MainMenu/MainMenu.react.js [hunk 5]
```diff
@@ -44,8 +45,26 @@ class MainMenu extends ConditionalRenderComponent
...
+        return Promise.all([
+            this.notifications.getNotificationCount(),
+            this.notifications.getNotifications(5)
+        ])
+        .then(values => this.setState({newNotifications : values[1], notificationCount : values[0]}));
```
Comment (kwierchris) [inline components/MainMenu/MainMenu.react.js]: To make it more readable I'd suggest the following:

.then([newNotifications, notificationCount]=> this.setState({newNotifications, notificationCount }))

File: components/MainMenu/MainMenu.react.js [hunk 6]
```diff
@@ -177,7 +196,7 @@ class MainMenu extends ConditionalRenderComponent
     render()
     {
         const { i18n, userData, router } = this.context;
-        const { activeMenuItem, newNotifications, recentNotifications } = this.state;
+        const { activeMenuItem, newNotifications,notificationCount } = this.state;
```
Comment (kwierchris) [inline components/MainMenu/MainMenu.react.js]: Space ;)

File: components/MainMenu/MainMenu.react.js [hunk 7]
```diff
@@ -225,19 +244,15 @@ class MainMenu extends ConditionalRenderComponent
...
+                        <NotificationsUI>
                             <div className="oc-notifications__header">{i18n.getMessage('MainMenu.newNotifications')}</div>
                             {
                                 newNotifications && newNotifications.length ?
-                                <Notification
-                                    svg={this.getIcon('info')}
-                                    svgClassName="fill-info"
-                                    message={<span>Your password will expire in 14 days. <a href="#">Change it now</a></span>}
-                                    dateTime="20/02/2017"/>
+                                newNotifications.map(item => (<Notification key={item.id} data-key={item.id} onClick={this.notificationClicked.bind(this,item.id)} svg={this.getIcon('info')} svgClassName="fill-info" message={item.title} dateTime={i18n.formatDateTime(item.changedOn)}/>))
```
Comment (kwierchris) [inline components/MainMenu/MainMenu.react.js]: TBH I am not a big fan of binding methods that are already part of the object again to the same object. Especially not when there are parameters passed. I know this is almost the same but from a readability point of view I'd really prefer **onClick={() => this.notificationClicked(item.id)}** here

File: components/MainMenu/MainMenu.react.js [hunk 8]
```diff
@@ -44,8 +45,26 @@ class MainMenu extends ConditionalRenderComponent
...
+
+    loadNotifications(){
+        return Promise.all([
+            this.notifications.getNotificationCount(),
+            this.notifications.getNotifications(5)
+        ])
+        .then(values => this.setState({newNotifications : values[1], notificationCount : values[0]}));
+    }
+
+    notificationClicked(args,sender){
```
Comment (kwierchris) [inline components/MainMenu/MainMenu.react.js]: This has a different brace style than there is in the rest of the file.

File: api/index.js [hunk 9]
```diff
@@ -1,5 +1,6 @@
 import Auth from './Auth';
 import Users from './Users';
+import Notifications from './Notifications';
 
-export { Auth, Users };
-export default { Auth, Users };
+export { Auth, Users,Notifications };
```
Comment (kwierchris) [inline api/index.js]: Spaces

File: api/Notifications.js [hunk 10]
```diff
@@ -0,0 +1,30 @@
...
+const {ApiBase} = require('./ApiBase');
+
+
+class Notifications extends ApiBase {
+    constructor() {
+        super();
+    }
+
+    getNotifications(count) {
+        var url = count?'/notification/api/deliveries?limit='+count:'/notification/api/deliveries';
```
Comment (kwierchris) [inline api/Notifications.js]: **const** would be better here.
And please add spaces too.

**Pull request closed without merging**